### DISCUSSION TITLE
Welcome to sahilk027.blog Discussions!

### DISCUSSION THREAD

**Discussion by SahilK-027** [SahilK-027/sahilk027.blog#1] [Announcements]
## 👋 Welcome!
I am thrilled to have you here! This space is designed to foster connections within our community. Feel free to engage in a variety of discussions, including:
  * Questions and Answers: Whether you're seeking insights, advice, or clarification, this is the perfect place to ask.
  * Brainstorming Sessions: Share your ideas for upcoming blog posts or collaborate with others to generate new and exciting content.
  * Community Engagement: Connect with fellow members, offer support, and celebrate each other's successes.
   *Open Dialogue: Embrace diverse perspectives and engage in respectful conversations. Remember, this community is built on mutual respect and understanding.  💪

To kick things off, introduce yourself in the comments below! Share a bit about who you are and what brings you to our community. We can't wait to get to know you better and embark on this journey together! 💬✨

**Comment by Chetan-KK** [comment]
Hayy! Saw your projects and stuff, it's pretty cool!

**Reply by SahilK-027** [reply]
Hey, thank you!

**Comment by captain17codes** [comment]
Hello Sahil! Remember when you came to RSCOE last time, we had a really long chat. Wanted to reconnect and also needed some quick help if you have time. I'm building a Godavari flood digital twin project (using real elevation + historical flood data), trying to show 3D buildings + rising water but currently struggling with MapLibre for that. You've worked a lot with Three.js/simulations, so thought I'd ask you, even a quick suggestion or you taking some time to guide me would help a lot. Let me know when you're free ..

**Reply by SahilK-027** [reply]
Hey!
I haven't actually worked with MapLibre myself; my background's more on the glsl-shader/Three.js/simulation side. So I can't speak to MapLibre-specific stuff.

But if you're open to it, here's an alternative angle: if the water simulation is the hard part (IMO realistic fluid simulation involves a lot of complexity which makes it hard), you could handle terrain + water entirely in Three.js load your data as a heightmap mesh, load satellite/map imagery as a texture on it for the colourmap, then do the buildings as extruded geometry and the rising water as an animated plane/shader (could even do a proper wave/flow shader if you want it to look more realistic than a flat rising plane). You'd skip MapLibre's map layer entirely and treat it as a pure 3D scene, which plays a lot more to what I actually know.

Let me know what you think!

**Reply by captain17codes** [reply]
That actually makes a lot of sense, let's go with that approach! I already have a DEM (elevation) file, building footprints as GeoJSON, and 4 precomputed flood extent levels (0.5m/1m/2m/3m rise) also as GeoJSON. I'll convert all of this to local X/Y/Z coordinates and export a heightmap + textures, then set up the terrain mesh, extrude the buildings, and work on the water shader. Will ping you if I get stuck somewhere.
